### ISSUE TITLE
Incorrect help message for excessive_precision lint

### ISSUE THREAD

**Issue by flip1995** [rust-lang-nursery/rust-clippy#3180]
As pointed out in https://github.com/rust-lang-nursery/rust-clippy/issues/2715#issuecomment-421145496 adding a `.` to a literal will trigger the excessive_precision lint and suggests to remove the `.`. This can lead to wrong code though:
```rust
fn main() {
    let y = 10f32;
    let x = y * 1_000_000_000.;
    println!("{}", x);
}
```

Will suggest
```
warning: float has excessive precision
 --> src/main.rs:3:17
  |
3 |     let x = y * 1_000_000_000.;
  |                 ^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `1_000_000_000`
  |
  = note: #[warn(clippy::excessive_precision)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#excessive_precision
```

which would lead to a type error. [Playground](https://play.rust-lang.org/?gist=b75290cfd7af11e2aee1cee6d4fe18f7&version=nightly&mode=debug&edition=2015)

I would suggest to just not lint in this case, just like `.0` is not linted.
